code/modules/vehicles/pimpin_ride.dm

```diff
@@ -0,0 +1,95 @@
+
+
+//PIMP-CART
+/obj/vehicle/janicart
+	name = "janicart (pimpin' ride)"
+	desc = "A brave janitor cyborg gave its life to produce such an amazing combination of speed and utility."
+	icon_state = "pussywagon"
+	keytype = /obj/item/key/janitor
+	var/obj/item/weapon/storage/bag/trash/mybag = null
+	var/floorbuffer = 0
+
+
+/obj/vehicle/janicart/handle_vehicle_offsets()
+	..()
+	if(buckled_mob)
+		switch(buckled_mob.dir)
+			if(NORTH)
+				buckled_mob.pixel_x = 0
+				buckled_mob.pixel_y = 4
+			if(EAST)
+				buckled_mob.pixel_x = -12
+				buckled_mob.pixel_y = 7
+			if(SOUTH)
+				buckled_mob.pixel_x = 0
+				buckled_mob.pixel_y = 7
+			if(WEST)
+				buckled_mob.pixel_x = 12
+				buckled_mob.pixel_y = 7
+
+
+/obj/item/key/janitor
+	desc = "A keyring with a small steel key, and a pink fob reading \"Pussy Wagon\"."
+	icon_state = "keyjanitor"
+
+
+/obj/item/janiupgrade
+	name = "floor buffer upgrade"
+	desc = "An upgrade for mobile janicarts."
+	icon = 'icons/obj/vehicles.dmi'
+	icon_state = "upgrade"
+
+
+/obj/vehicle/janicart/Move(a, b, flag)
+	..()
+	if(floorbuffer)
+		var/turf/tile = loc
+		if(isturf(tile))
+			tile.clean_blood()
+			for(var/A in tile)
+				if(istype(A, /obj/effect))
+					if(is_cleanable(A))
+						qdel(A)
```

Comment:
I made the Moved() proc just to handle this kind of shit. And also to kill those terrible arg names which are 100% incorrect.